### PR TITLE
[WIP] Rewiring

### PR DESCRIPTION
**Pull request by ambaxter**

### CODE REVIEW

File: libsolv-sys/build.rs [hunk 1]
```diff
@@ -1,7 +1,67 @@
...
+        .file("static/bitmap.c")
+        .file("static/dirpool.c")
+        .file("static/pool.c")
+        .file("static/poolarch.c")
+        .file("static/repo.c")
+        .file("static/repodata.c")
+        .file("static/strpool.c")
+        .compile("libsolv-static-functions.a");
+
+    println!("cargo:rustc-link-lib=solv");
```
Comment (igor-raits) [inline libsolv-sys/build.rs]: Why not pkgconfig?

File: libsolv-sys/src/export/strpool.rs [hunk 2]
```diff
@@ -0,0 +1,6 @@
+use {Stringpool, Id};
+use libc::c_char;
+
+extern "C" {
+    pub fn e_stringpool_id2str(ss: *mut Stringpool, id: Id) -> *const c_char;
+}
```
Comment (igor-raits) [inline libsolv-sys/src/export/strpool.rs:6]: Newline please

File: libsolv-sys/static/pool.c [hunk 3]
```diff
@@ -0,0 +1,17 @@
+#include <solv/pool.h>
+
+Solvable *e_pool_id2solvable(const Pool *pool, Id p) {
```
Comment (igor-raits) [inline libsolv-sys/static/pool.c:3]: Keep one code style... Return type is one line, name and args is next

File: libsolv-sys/build.rs (hunk 1 above)
Comment (ambaxter) [reply libsolv-sys/build.rs]: It apparently didn't need it

File: src/solver.rs [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+            unsafe {solver_create(borrow._p)}
+        };
+        Solver{ctx: ctx, _s: _s}
+    }
+
+    pub(crate) fn new_with_solver(ctx: Rc<RefCell<Pool>>, _s: *mut _Solver) -> Self {
+        Solver{ctx: ctx, _s: _s}
+    }
+
+    pub fn solve(&mut self, job: &mut Queue) -> libc::c_int {
```
Comment (ambaxter) [inline src/solver.rs:27]: +    pub fn solve(&mut self, job: &mut Queue) -> libc::c_int {
ignatenkobrain
it should rather return vec of Problem's
ignatenkobrain
yeah, it's mostly wrapping stuff, but would be awesome to do that :D
abaxter
describe what you'd like it to do in detail. I'm up for making changes
abaxter
I mean, there's not much there
ignatenkobrain
so, solve() returns number of problems which you can analyze later out of Solv* object.
ignatenkobrain
in all bindings, it returns list of Porblem objects
ignatenkobrain
typedef struct {                         Solver *solv;                         Id id;                       } Problem;
ignatenkobrain
https://github.com/openSUSE/libsolv/blob/master/bindings/solv.i#L3045
ignatenkobrain
so basically it creates Problem object which has back reference to Solver and problem ID
ignatenkobrain
and there are some solv_*() methods which work with those problems
ignatenkobrain
I will be back in  few hours, taking a small bike trip
abaxter
k
ignatenkobrain
abaxter: we will need to discuss features in ext part
ignatenkobrain
once I will be back
ignatenkobrain
abaxter: https://github.com/fedora-modularity/depchase/blob/master/depchase#L363-L368

File: src/ext/rpmmd.rs [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+pub trait RpmMd {
+
+    fn add_repomd(&mut self, file: & File) -> bool;
+
+    fn add_repomdxml(&mut self, file: & File) -> bool;
+}
+
+impl RpmMd for Repo {
+    fn add_repomd(&mut self, file: &File) -> bool {
```
Comment (igor-raits) [inline src/ext/rpmmd.rs]: I would rather return Result

Comment (ambaxter) [reply src/ext/rpmmd.rs]: That is the plan, but right now I'm trying to get all the way through depchase before I start refactoring

Comment (ambaxter) [reply src/ext/rpmmd.rs]: Check it out, now?
